gradle-plugin/src/proguard/gradle/ProGuardTask.java

```diff
@@ -63,36 +67,45 @@
     private static final String CONFIGURATION_FILE_NAME_PREFIX      = "/proguard/gradle/proguard-";
     private              String resolvedConfigurationFileNamePrefix = getProject().file(CONFIGURATION_FILE_NAME_PREFIX).toString();
 
+    @Inject
+    protected abstract ObjectFactory getObjectFactory();
+
+    @Nested
+    protected ProGuardTaskConfiguration getTaskConfiguration() {
+        return ProGuardTaskConfiguration.create(getConfigurationProvider(), getObjectFactory());
+    }
```

Comment:
I think this needs to be public. Otherwise there is no way to correctly depend on task outputs like the mapping file. My use case is publishing the mapping file to a repository and I want to do something like this:

```kotlin
publishing {
    publications {
        create<MavenPublication>("sourcemap") {
            groupId = "com.example"
            artifactId = "my-app-sourcemap"
            version = "$version"
            artifact(tasks.proguard.taskConfiguration.printMapping)
        }
    }
}
```

The task dependency to ProGuard should automatically be set up.

Reply:
You're right. This needs to be `public`. Done.